src/models/room.ts

```diff
@@ -2199,8 +2199,14 @@ export class Room extends EventEmitter {
      *                   message events into the room.
      */
     public maySendMessage(): boolean {
-        return this.getMyMembership() === 'join' &&
-            this.currentState.maySendEvent(EventType.RoomMessage, this.myUserId);
+        if (this.getMyMembership() !== "join") return false;
+
+        if (this.client.isRoomEncrypted(this.roomId)) {
+            return (this.client.crypto || this.client.usingExternalCrypto) &&
```

Comment:
I actually think `(this.client.crypto || this.client.usingExternalCrypto)` may be redundant and you were right initially, as js-sdk won't let you send unencrypted content in encrypted rooms (unless usingExternalCrypto but that's special because we trust the crypto service to encrypt for us)

So I think reverting back to your first solution is sane.